tests/worker/test_reworker.py

```diff
@@ -7,106 +8,236 @@
+    task_ids = [worker.submit_task(_SIMPLE_TASK) for _ in range(num_runs)]
+    event_sequences = [_sleep_events(task_id) for task_id in task_ids]
+
+    for task_id, events in zip(task_ids, event_sequences):
+        assert_run_produces_worker_events(events, worker, task_id)
+
+
+def _sleep_events(task_id: str) -> List[WorkerEvent]:
+    return [
+        WorkerEvent(
+            state=WorkerState.RUNNING,
+            task_status=TaskStatus(
+                task_id=task_id, task_complete=False, task_failed=False
             ),
-            WorkerEvent(
-                state=WorkerState.IDLE,
-                task_status=TaskStatus(
-                    task_name="test", task_complete=False, task_failed=False
-                ),
-                errors=[],
-                warnings=[],
+            errors=[],
+            warnings=[],
+        ),
+        WorkerEvent(
+            state=WorkerState.IDLE,
+            task_status=TaskStatus(
+                task_id=task_id, task_complete=False, task_failed=False
             ),
-            WorkerEvent(
-                state=WorkerState.IDLE,
-                task_status=TaskStatus(
-                    task_name="test", task_complete=True, task_failed=False
-                ),
-                errors=[],
-                warnings=[],
+            errors=[],
+            warnings=[],
+        ),
+        WorkerEvent(
+            state=WorkerState.IDLE,
+            task_status=TaskStatus(
+                task_id=task_id, task_complete=True, task_failed=False
             ),
-        ],
-        worker,
+            errors=[],
+            warnings=[],
+        ),
+    ]
+
+
+def test_no_additional_progress_events_after_complete(worker: Worker):
+    """
+    See https://github.com/bluesky/ophyd/issues/1115
+    """
+
+    progress_events: List[ProgressEvent] = []
+    worker.progress_events.subscribe(lambda event, id: progress_events.append(event))
+
+    task: Task = RunPlan(
+        name="move", params={"moves": {"additional_status_device": 5.0}}
     )
+    task_id = worker.submit_task(task)
+    begin_task_and_wait_until_complete(worker, task_id)
+
+    # Extract all the display_name fields from the events
+    list_of_dict_keys = [pe.statuses.values() for pe in progress_events]
+    status_views = [item for sublist in list_of_dict_keys for item in sublist]
+    display_names = [view.display_name for view in status_views]
+
+    assert "STATUS_AFTER_FINISH" not in display_names
+
+
+#
+# Worker helpers
+#
+
+
+def assert_run_produces_worker_events(
+    expected_events: List[WorkerEvent],
+    worker: Worker,
+    task_id: str,
+) -> None:
+    assert begin_task_and_wait_until_complete(worker, task_id) == expected_events
 
 
-def submit_task_and_wait_until_complete(
-    worker: Worker, task: Task, timeout: float = 5.0
+def begin_task_and_wait_until_complete(
+    worker: Worker,
+    task_id: str,
+    timeout: float = 5.0,
```

Comment:
We're going to continue adding more running of plans into tests. Some of those plans/tests will involve waiting, sleeping or checking time to know that a wait or a sleep has been done. In order to prevent this from inflating the time it takes to run our tests, we should use this early opportunity to make tests that mock out time and sleep to have simple examples to point to and to build off of. This also means that our tests will be more robust against external slowness of compute eating up fractions of our timeouts.

[Something like this](https://stackoverflow.com/questions/57747998/how-to-make-unit-test-with-asyncio-sleep-contained-code)

Reply:
I feel this is a scope creep issue; I agree it is a good idea to do this, we should make a separate PR.

Reply:
I agree: creep bad!